plugins/tactics/src/Ide/Plugin/Tactic.hs

```diff
@@ -95,7 +96,9 @@ commandProvider Intros =
     provide Intros ""
 commandProvider Split =
   foldMapGoalType (F.fold . tyDataCons) $ \dc ->
-    provide Split $ T.pack $ occNameString $ getOccName dc
+    let cname = occNameString $ getOccName dc
+    in  requireExtensionWhen MagicHash ("#" `L.isSuffixOf` cname) $
```

Comment:
Let's use the `isAlgebraic` function to do this check instead, though I'd be open to moving this check there.

Reply:
https://github.com/isovector/haskell-language-server/blob/8772f629191816b19abd73a3fceef2210570e93a/plugins/tactics/src/Ide/Plugin/Tactic/GHC.hs#L49

Reply:
Oh sorry, I misunderstood the check here. I'd argue this behavior is so far out of the ordinary experience of most devleopers that it's not worth supporting.

Reply:
For the moment, I will use the `algebraicTyCon` check, add an ignored test case refering to #31 and tackle it in a follow up PR.

Reply:
Sounds good to me. I really appreciate your keen eye on this!